web-common/src/features/scheduled-reports/utils.ts

```diff
@@ -151,8 +170,10 @@ export function getUpdatedAggregationRequest(
   aggregationRequest: V1MetricsViewAggregationRequest,
   filtersArgs: FiltersState,
   timeControlArgs: TimeControlState,
+  rows: string[],
+  columns: string[],
   exploreSpec: V1ExploreSpec,
-) {
+): V1MetricsViewAggregationRequest {
```

Comment:
Given all the parameters, maybe this should function take a parameter object too?

Reply:
Additionally/alternatively, it may be a nice pipeline if this function were implemented as:
```ts
let aggRequest: ...;
aggRequest = updateAggregationRequestWithTimeRange(aggRequest, ...)
aggRequest = updateAggregationRequestWithRowsAndColumns(aggRequest, ...)
aggRequest = updateAggregationRequestWithFilters(aggRequest, ...)
```
Something like that would be easier to follow.

Reply:
Good idea. I have been thinking of adding a builder for aggregation request for a while. I guess this is a good time to start. We can refactor other places to use this builder so that all the code is in a common place.